shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -57,30 +58,56 @@ func NewSequenceDispatcher(
 }
 
 func (sd *SequenceDispatcher) Add(queueItem models.QueueItem) error {
-
 	if sd.mode == common.SDModeRW {
 		//if there is only one shipyard we can both read and write,
-		//so we try to dispatch the sequence immediately
-		if err := sd.dispatchSequence(queueItem); err != nil {
-			if errors.Is(err, ErrSequenceBlocked) {
-				//if the sequence is currently blocked, insert it into the queue
-				return sd.add(queueItem)
-			} else if errors.Is(err, ErrSequenceBlockedWaiting) {
-				//if the sequence is currently blocked and should wait, insert it into the queue
-				if err2 := sd.add(queueItem); err2 != nil {
-					return err2
+		//so we try to dispatch the sequence immediately if the sequence queue is empty
+		queueEmpty, err := sd.isQueueEmpty(queueItem)
+		if err != nil {
+			return err
+		}
+		if queueEmpty {
+			if err := sd.dispatchSequence(queueItem); err != nil {
+				if errors.Is(err, ErrSequenceBlocked) {
+					//if the sequence is currently blocked, insert it into the queue
+					return sd.add(queueItem)
+				} else if errors.Is(err, ErrSequenceBlockedWaiting) {
+					//if the sequence is currently blocked and should wait, insert it into the queue
+					return sd.addItemToQueue(queueItem)
+				} else {
+					return err
 				}
-				return ErrSequenceBlockedWaiting
-			} else {
-				return err
 			}
+			return nil
+		} else {
+			//if there are sequences in queue, insert into queue
+			return sd.addItemToQueue(queueItem)
 		}
-		return nil
 	} else {
 		//if there are multiple shipyard we should only write
 		return sd.add(queueItem)
 	}
+}
 
+func (sd *SequenceDispatcher) addItemToQueue(queueItem models.QueueItem) error {
+	if err2 := sd.add(queueItem); err2 != nil {
+		return err2
+	}
+	return ErrSequenceBlockedWaiting
+}
+
+func (sd *SequenceDispatcher) isQueueEmpty(queueItem models.QueueItem) (bool, error) {
+	queuedSequences, err := sd.sequenceQueue.GetQueuedSequences()
+	if err != nil {
+		if !errors.Is(err, db.ErrNoEventFound) {
+			return false, sd.addItemToQueue(queueItem)
```

Comment:
calling `addItemToQueue()` within the `isQueueEmpty` seems counterintuitive - can we move that outside?